Update create-and-link-connections.md (#3291)
astronomer/docs: astro/create-and-link-connections.md

You can create and manage Airflow connections for Deployments with the Astro Environment Manager in the Cloud UI. The Environment Manager uses an Astro-managed secrets backend to store connection configurations as Kubernetes Secrets.

> :::info Note
> 
> Astro Environment Manager is only available to Astro Hosted users; it is not available for Astro Hybrid users.
> 
> 

Reviewer: ```suggestion
:::info

The Astro Environment Manager is available only on Astro Hosted

:::
```

Reviewer: This is smaller, but it might be more visually distinctive and therefore visible for readers

Author: I'm not opposed to the hosted badge, but should it say something more obvious like "Hosted Only"? I'm not sure a Hybrid user would find the badge to obviously imply it's hosted-only.

Reviewer: Good point, we can work on creating a separate "Hosted only" badge in the next quarter to apply on docs.
